### PR TITLE
Add container registry types

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
According to our discussion, I have added container registry types into the base src of ZenML, in order that we can validate container registries in a nice case-by-case basis.

I am looking for either @schustmi or @stefannica to take over this branch at this point. I feel like you might take it to completion easier, and I have done all the copy paste work here. If you guys are busy, also happy to take this into completion. Some feedback here would be welcome in this case :100: 

# TODO:

- [ ] Update docs (we can do this in @AlexejPenner branch)
- [ ] Write tests

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/features/integrations) table.
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/enums.py [hunk 1]
```diff
@@ -73,3 +73,15 @@ class StoreType(StrEnum):
     LOCAL = "local"
     SQL = "sql"
     REST = "rest"
+
+
+class ContainerRegistry(StrEnum):
```
Comment (schustmi) [inline src/zenml/enums.py:78]: I think in all other cases we handle these flavors as simple string constants now, but I can see the point of having them as an enum with that many zenml core implementations. I'd still vote for changing the enum name to `ContainerRegistryFlavor` or similar

File: src/zenml/enums.py [hunk 2]
```diff
@@ -73,3 +73,15 @@ class StoreType(StrEnum):
...
+    """Types of container registries."""
```
Comment (schustmi) [inline src/zenml/enums.py:79]:
```suggestion
    """Flavors of container registries."""
```

File: src/zenml/container_registries/github_container_registry.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from typing import ClassVar
+
+from zenml.container_registries.base_container_registry import (
+    BaseContainerRegistry,
+)
+from zenml.enums import ContainerRegistry, StackComponentType
+
+
+class GitHubContainerRegistry(BaseContainerRegistry):
+    """Class for ECR Container Registry.
```
Comment (schustmi) [inline src/zenml/container_registries/github_container_registry.py:23]:
```suggestion
    """Class for Github Container Registry.
```

File: src/zenml/container_registries/aws_container_registry.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    Attributes:
+        uri: The URI of the container registry.
+    """
+
+    # Class Configuration
+    TYPE: ClassVar[StackComponentType] = StackComponentType.CONTAINER_REGISTRY
+    FLAVOR: ClassVar[str] = ContainerRegistry.AWS
+
+    @validator("uri")
+    def validate_aws_uri(cls, uri: str) -> None:
```
Comment (schustmi) [inline src/zenml/container_registries/aws_container_registry.py:36]: I really like the validation with the error message giving a concrete example! One thing that might be worth considering is allowing a trailing slash and simply removing it? And probably we can even have a regex that checks if the uri ends with `amazonaws.com`, but I'm not familiar enough with the potential urls they might have

**Review by schustmi — CHANGES_REQUESTED** [review]
Very nice change that aligns the container registries with all the remaining stack components. I left a few minor comments, and will add two more general requests here:
- I think all container registries would benefit from receiving additional validations with example URLs similar to the one you wrote for AWS
- I'm guessing we can/should update some docs here to use the new container registry flavors? While using the default one would still work, I think our examples (e.g. running on EKS kubeflow) would benefit from the additional validation that it's an ECR registry

**Review by stefannica — APPROVED** [review]

**Pull request merged**